package.json

```diff
@@ -42,6 +44,13 @@
     "z-schema-errors": "^0.2.1"
   },
   "devDependencies": {
+    "@babel/cli": "^7.12.8",
+    "@babel/core": "^7.12.9",
+    "@babel/eslint-parser": "^7.12.1",
+    "@babel/node": "^7.12.6",
+    "@babel/plugin-proposal-optional-chaining": "^7.12.7",
+    "@babel/preset-env": "^7.12.7",
+    "babel-eslint": "^10.1.0",
```

Comment:
Doesn't that kinda defeats the purpose of semantic versioning? Seems like we would want to bring in patches automatically.

In practice, package-lock only adds value by ensuring consistent deploys between environments, so I'm not sure what benefit we get here by tracking it in master.

Reply:
Since we commit package lock, the same dependency versions will always be used, unless the lockfile is updated. Having exact versions in package.json helps quickly determine (if needed) what versions we're using without looking into the lockfile. For production (or non-dev) dependencies it's a bit different, as the lockfile is not part of published packages, and using version ranges lets npm install the latest compatible version and/or dedupe dependencies.

This is a matter of preference I guess, but that's why existing dev dependencies have exact versions.

Reply:
> what versions we're using without looking into the lockfile

the lockfile does not indicate the versions of the installed packages. You have to look at the version property in package.json for the dependency inside of `node_modules`. e.g. consider the case where you jump to a different commit (via pull, checkout, bisect, etc...) that has different dependencies (or versions of dependencies) set in package.json, and maybe you forgot to run `npm install` after the checkout.

> using version ranges lets npm install the latest compatible version and/or dedupe dependencies

Indeed, and it's really nice when this can also happen for devDependencies. I've learned this the hard way a few times over.

In order to keep the discussion on topic, I'll go ahead and remove the carets from the code in the branch for this PR, and we can figure out a good setting for [`save-*` in the npm config](https://docs.npmjs.com/cli/v6/using-npm/config#save-exact) in a separate PR / issue thread